src/bucket/test/BucketManagerTests.cpp

```diff
@@ -1280,7 +1280,8 @@ class StopAndRestartBucketMergesTest
                         "Switching protocol at ledger {} from protocol {} "
                         "to protocol {}",
                         i, firstProtocol, secondProtocol);
-                    cfg.LEDGER_PROTOCOL_VERSION = secondProtocol;
+                    cfg.TESTING_UPGRADE_LEDGER_PROTOCOL_VERSION =
```

Comment:
@graydon (or maybe @marta-lokhova)

I was just reviewing this change. The log line on 1278-1282 suggests that this should induce a protocol upgrade. But it doesn't look like it will, and testing indicates that it does not. What is actually supposed to happen here? Changing behavior here is outside the scope of this PR, so if there is work to be done then I think we should open a new issue to track it.

If there is an issue here, then it's also probably worth auditing that the rest of the tests do work as expected.

Reply:
from what I can tell, this is a bug. The comment on lines 688-693 suggests that we need to upgrade to verify that bucketlist behaves as expected, but, like you said, we're not actually doing this properly in this test.

Reply:
Thanks for confirming @marta-lokhova! I'll open an issue to track this.